tox.ini

```
@@ -173,6 +173,10 @@ allowlist_externals =
 deps =
     towncrier
     -r{toxinidir}/docs/requirements.txt
-commands =
+commands_pre =
     towncrier --version
-    {envpython} -c 'import sys, subprocess; subprocess.run([sys.executable, "-m", "towncrier", "build", "--version", "main", "--draft"], stderr=subprocess.PIPE)'
+# wrap the invocation of `towncrier build --version main --draft` to discard stderr
+# unfortunately, stderr gets interleaved with stdout, often mixing towncrier's draft build output
+# with user messaging about the invocation
+commands =
+    {envpython} -bb -I -c 'import sys, subprocess; subprocess.run([sys.executable, "-bb", "-I", "-m", "towncrier", "build", "--version", "main", "--draft"] + sys.argv[1:], stderr=subprocess.DEVNULL)' {posargs}
```

Did the `-E -s` bits not work for some reason? https://github.com/jazzband/pip-tools/pull/2203#discussion_r2222207208

---

Ah, okay. This was copied from older projects that used to support huge ranges of Python versions (down to Python 2). I should probably update this in my projects, then. I think that some of the strictness options used to break within Click but couldn't remember which ones. I think, this was `-bb`.

That said, we forgot to start catching warnings early. Let's try this
```suggestion
    {envpython} -bb -I -Werror -c 'import sys, subprocess; subprocess.run([sys.executable, "-bb", "-I", "-Werror", "-m", "towncrier", "build", "--version", "main", "--draft", *sys.argv[1:]], stderr=subprocess.DEVNULL)' {posargs}
```